### ISSUE TITLE
allow setting content-type when calling `view(template, obj)`

### ISSUE THREAD

**Issue by DarthGoon** [fastify/point-of-view#9]
hey there- I recently started playing around with Fastify, so excuse me if i just missed something before making this issue.

I have a use case where i want to output xml (rss feed).  The rss template is actually in a pug template.  This is just because it's easier to work with.
So the flow is--  I go get my data, set the content-type with `reply.header('Content-Type', 'text/xml');` then call `view()` with the template and my data.

The expectation is that i'll end up with an xml response.  What actually happens is that the content-type gets overridden when you call `view()` with a template-
https://github.com/fastify/point-of-view/blob/master/index.js#L53

I'm curious if there's a specific reason behind this?  Or if it's acceptable to check for a content-type defined, and not assign `text/html` if there is one?

**Comment by DarthGoon** [comment]
```  res
    .header('Content-Type', 'text/xml')
    //.send('<thing jhsdf="124">kjnvuhr</thing>');
    .view('rss.pug', { a bunch of data });
```

in the above code-  I will get back a `text/html` response calling `view`.  But if you swap the commented `send` with `view`, you'll get back an xml response.

**Comment by delvedor** [comment]
Hi!
Thank you for using fastify :)

Yes, we are setting the content type on `'text/html'` because internally Fastify (*if you didn't set a content-type*) assumes that you are sending an `'application/json'`.
So to supply a better usability of this module we decided to automatically add the content type, so you don't need to do:
```js
reply.header('Content-Type').view(template.pug, { data: 'data' })
```
Honestly I didn't think about your use case :)

Your suggestion is the actual solution:
```js
if (!reply.res.getHeader('Content-Type')) {
  reply.header('Content-Type', 'text/html')
}
```
If you want you can do a PR, if not I'll do it :)

**Comment by DarthGoon** [comment]
cool... i made a PR - https://github.com/fastify/point-of-view/pull/10

**Comment by delvedor** [comment]
Closed by #10
